api/src/routers/meals.js

```diff
@@ -0,0 +1,127 @@
+import express from "express";
+import knex from "../database_client.js";
+
+const router = express.Router();
+
+router.get("/meals", async (req, res) => {
+  try {
```

Comment:
In this endpoint we check if the request by using one large try catch, I think would be better for the person calling the endpoint to be informed why their request didn't succeed.

For example:
```
if (req.query.limit && /^-?\d+$/.test(str.trim()))
{
    ...
}else{
    res.status(400).json({ error: "req.query value: "+req.query.limit+" is not a valid limit, most be an integer"});
    return;
}
```